### PR TITLE
WEBUI-71: Hide dropzone for read-only users

### PR DESCRIPTION
**Pull request by charlesboidot**
Fix for Jira Issue: https://jira.nuxeo.com/browse/WEBUI-71

Test And Push[In Progress]: https://qa2.nuxeo.org/jenkins/job/WebUI/job/web-ui-pipeline-9.10/422/

### CODE REVIEW

File: elements/nuxeo-document-viewer/nuxeo-document-viewer.html [hunk 1]
```diff
@@ -57,9 +63,19 @@
...
     is: 'nuxeo-document-viewer',
-    behaviors: [Nuxeo.I18nBehavior],
+    behaviors: [Nuxeo.I18nBehavior, Nuxeo.FiltersBehavior],
     properties: {
       document: Object
-    }
+    },
+
+    _thumbnail: function(doc) {
+      return doc && doc.uid && doc.contextParameters && doc.contextParameters.thumbnail &&
```
Comment (andreacornaglia) [inline elements/nuxeo-document-viewer/nuxeo-document-viewer.html:72]: for easier reading, could simplify to && doc.contextParameters.thumbnail.url || ''; (so no need for the ternary operator)

File: elements/nuxeo-document-viewer/nuxeo-document-viewer.html [hunk 2]
```diff
@@ -57,9 +63,19 @@
...
+      doc.contextParameters.thumbnail.url ? doc.contextParameters.thumbnail.url : '';
+    },
+
+    _isDropzoneAvailable: function (doc) {
+      return doc && this.hasPermission(doc, 'WriteProperties') &&
```
Comment (andreacornaglia) [inline elements/nuxeo-document-viewer/nuxeo-document-viewer.html:77]: does it make more sense to put doc.type !== 'Root' before the functions?

File: elements/nuxeo-document-viewer/nuxeo-document-viewer.html [hunk 3]
```diff
@@ -57,9 +63,19 @@
...
+        !this.isImmutable(doc) && doc.type !== 'Root' && !this.isTrashed(doc);
+    },
```
Comment (RSalem07) [inline elements/nuxeo-document-viewer/nuxeo-document-viewer.html:79]: for `_isDropzoneAvailable ` ir can be done as:

```
return this.hasPermission(doc, 'WriteProperties') 
       !this.isImmutable(doc) &&
       !this.hasType(doc, 'Root') &&
       !this.isTrashed(doc)
```
I mean the nullity check on `doc` is already done by `hasPermission` and we have a helper method on the filter behaviour for checking document type

File: elements/nuxeo-document-viewer/nuxeo-document-viewer.html [hunk 4]
```diff
@@ -57,9 +63,19 @@
...
     properties: {
       document: Object
-    }
+    },
+
+    _thumbnail: function(doc) {
+      return doc && doc.uid && doc.contextParameters && doc.contextParameters.thumbnail &&
+      doc.contextParameters.thumbnail.url ? doc.contextParameters.thumbnail.url : '';
+    },
+
```
Comment (RSalem07) [inline elements/nuxeo-document-viewer/nuxeo-document-viewer.html:75]: As Andrea suggested, you can avoid the ternary operator and use:

```
return (doc && doc.uid 
       && doc.contextParameters 
       && doc.contextParameters.thumbnail 
       && doc.contextParameters.thumbnail.url) || '';
```

**Review by RSalem07 — COMMENTED** [review]
Thanks for the work 👍, I left some comments / suggestions

**Comment by RSalem07** [comment]
@charlesboidot I don't know if you now that we have to launch a qa2 jenkins job: https://qa2.nuxeo.org/jenkins/job/WebUI/job/web-ui-pipeline-9.10/ with your branch.

**Review by RSalem07 — APPROVED** [review]

**Review by andreacornaglia — APPROVED** [review]

**Comment by charlesboidot** [comment]
Backport from 10.10 of https://github.com/nuxeo/nuxeo-web-ui/blob/10.10/elements/nuxeo-document-viewer/nuxeo-document-viewer.html

**Pull request merged**